pkg/controllers/metrics/pod/controller.go

```diff
@@ -107,14 +126,24 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 		return reconcile.Result{}, err
 	}
-	c.record(ctx, pod)
+	c.record(ctx, pod, ts)
 	return reconcile.Result{}, nil
 }
 
-func (c *Controller) record(ctx context.Context, pod *v1.Pod) {
+func (c *Controller) record(ctx context.Context, pod *v1.Pod, timestamp time.Time) {
+	// Record pods state metric
 	labels := c.labels(ctx, pod)
 	podGaugeVec.With(labels).Set(float64(1))
 	c.labelsMap.Store(client.ObjectKeyFromObject(pod), labels)
+
+	// Record pods startup time metric
+	podKey := client.ObjectKeyFromObject(pod).String()
+	if pod.Status.Phase == phasePending {
+		c.pendingPods.Insert(podKey)
+	} else if c.pendingPods.Has(podKey) {
+		c.pendingPods.Delete(podKey)
+		podStartupTimeSummary.Observe(timestamp.Sub(pod.CreationTimestamp.Time).Seconds())
```

Comment:
Can you switch these two lines, we should delete the pod from the list after we observe it.